CMakeLists.txt

```diff
@@ -67,7 +67,11 @@ function(link_to_source base_name)
         if (CMAKE_HOST_UNIX)
             set(command ln -s ${target} ${link})
         else()
-            set(command cmd.exe /c mklink /j ${link} ${target})
+            if (IS_DIRECTORY ${target})
+                set(command cmd.exe /c mklink /j ${link} ${target})
+            else()
+                set(command cmd.exe /c mklink ${link} ${target})
```

Comment:
This line needs to be a hardlink, not a symbolic link, to avoid requiring local Admin privileges. e.g.:
```
set(command cmd.exe /c mklink /h ${link} ${target})
```

Reply:
I don't think a hard link is reasonable. A legitimate, realistic use case. is to have a read-only source tree (whether for permission or performance reasons) and to do the build on a different, read-write filesystem.

If a symbolic link is not possible, we should copy.

Reply:
I think that's the argument for #1496. What we want to do now is to get the build working again. Symbolic links are a regression, and will break the build on machines where it previously built.

What you're suggesting is reasonable, but not a regression, and therefore is an enhancement, which I think will be fixed by #1496.

I therefore see no reason to delay merging this fix assuming it passes the CI, for the sake of an enhancement.

cc: @Patater

Reply:
Changing to a hard link is a regression from #1470. It passes on the CI by luck because of how we set up the directory structure for the builds, just like symbolic links pass on the CI by luck because our build environment happens to have the required permission.